policy/container/gke_cos_image_used.rego

```diff
@@ -0,0 +1,74 @@
+# Copyright 2019 The resource-policy-evaluation-library Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+package gcp.container.projects.locations.clusters.policy.cos_image_used
```

Comment:
Should this check just happen at the nodepool level?

Reply:
Changing image used (whether via clusers or cluster.nodePools update) is treated as cluster upgrade, and can be the only running operation on cluster. 
During the image change, cluster ***status*** field is changed to ***UPGRADING***, but nodepool is in ***RUNNING*** status - moving check to nodepool instead of cluster would give us lots of 400 "resource not ready" errors.

Reply:
Btw, i submitted this as a bug on the google issue tracker: https://issuetracker.google.com/issues/139740490

Maybe if they fix it, we can move this policy to the nodepool resource